src/include/sof/timer.h

```diff
@@ -70,4 +70,6 @@ static inline uint64_t timer_get_system(struct timer *timer)
 	return arch_timer_get_system(timer);
 }
 
+extern struct timesource_data platform_generic_queue[];
```

Comment:
Why made this to a global variable?
You can take reference to other clock user like: `sch->clock` or just use `PLATFORM_DEFAULT_CLOCK`